src/kata6.joinNames.js

```diff
@@ -1,3 +1,25 @@
-const joinNames = (namesObj) => {};
+const joinNames = (namesObj) => {
+let names = namesObj.map((x) => x.name);
+//.split(" "));
+let names1 = String(names);
+return names1.split(",").join(", ").replace(/,(?=[^,]+$)/, " &");
```

Comment:
Theres some unnecessary complications here, you are creating an array from the object, then casting that array into a string only to split it back out into an array again to run the regex on it. You could remove two of those steps and reduce the noise somewhat:

```
  let names = namesObj.map((x) => x.name);
  return names.join(", ").replace(/,(?=[^,]+$)/, " &");
```
Note here the indentation and the commented out line 3 being removed too. Nice regex btw!